### DISCUSSION TITLE
Is trame suitable for building a website project with separated front-end and back-end?

### DISCUSSION THREAD

**Discussion by chiyandetaotie** [Kitware/trame#515] [Trame/VTK] [answered]
I wonder if "trame" is designed primarily for browser application building.Trame is a framework with integrated front-end and back-end.The trame backend is developed in python, while the front-end vue code is defined together in a python project. 
        For modern web development, flexibility and extensibility are necessary.  I want to build a browser application with a separating front-end and back-end. The project uses vue for the frontend and django for the backend. Trame is not separated from the front-end and back-end. So I want to use the vtk.js to build the frontend like the project glance ([https://github.com/Kitware/glance)](https://github.com/Kitware/glance).), but the vtk.js does not support all C++ functions and also does not support unstructured grid.
        Can trame be used to build platform service applications with separate front-end and back-end? 
        Can trame be integrated into Django?
        In this case, which technology stack should I use to implement the separating front-end and back-end browser application？trame or the vtk.js(like glance) or other method?

**Comment by jourdain** [comment, marked as answer]
Trame is mainly focused at bringing desktop like application to the web. In other words, creating desktop app with a web front-end that can run locally, in the cloud, jupyter or on HPC.

If your goal is to create a web-app that scale because most/all of the processing is done on the client side, trame is not for you. 

Trame can still be used with a front/back end separation, but that won't remove the fact that trame expect the processing to happen on the server side. 

For the django integration, a trame process could run as an asynchronous task. So if django can support `async` in python, then technically they can both coexist, but I would not suggest that in anyway. Think of it that way, would you want a VTK operation to kill your Django server just because you made a bad method call?

In term of technology stack, if you want more than vtk.js on the client side to do things, you should look into VTK/wasm.
